### ISSUE TITLE
Generated sitemaps are invalid, loc tags needs to be absolute

### ISSUE THREAD

**Issue by nikolowry** [Dolphiq/craft3-plugin-sitemap#7]
Per https://www.sitemaps.org/protocol.html, the `loc` tag definition is as followed:
> URL of the page. This URL must begin with the protocol (such as http) and end with a trailing slash, if your web server requires it. This value must be less than 2,048 characters.

However, the plugin is generating relative `loc` values, which results in errors in Google's Search Console:
![img-2018-01-12-183654](https://user-images.githubusercontent.com/597310/34899542-9841b482-f7c7-11e7-9e1e-85ee866be254.png)

---

A little off-topic, but would you happen to know why `/sitemap.xml` returns a 404 status on a NGINX/PHP-FPM stack even though it renders the XML? The only way I could submit a site map to Google's Search Console was by using `/?p=actions/sitemap/sitemap/index`
![img-2018-01-12-184153](https://user-images.githubusercontent.com/597310/34899669-4a1858d2-f7c8-11e7-8d52-0c9ae4b061ad.png)

**Comment by nikolowry** [comment]
I temporarily got the relative `<loc>` values by installing `nginx-full` on Ubuntu which has `ngx_http_substitutions_filter_module` enabled and then including this snippet near the top of my server block:
```
# Ensure absolute paths in Sitemap xml
sub_filter '<loc>/' '<loc>https://$host/';
sub_filter_once off;
sub_filter_types text/xml;
```

Concerning the 404 on `/sitemap.xml` -- I have a near identical setup locally as I do in production, the only difference being expires, cache and gzip headers configured for Google's Pagespeed. The `/sitemap.xml` works locally, but not on production.  Will continue to investigate

**Comment by johanzandstra** [comment]
Hi @nikolowry The /?p=actions/sitemap/sitemap/index action should work also the same way as /sitemap.xml

The plugin fetches the base url from the table sites.baseUrl can you verify that the value is set for the site? The value should be set if you set it in the Craft main settings sites area.

**Comment by nikolowry** [comment]
@johanzandstra ah ha -- therein lies our problem! Our sites are always coded to dynamically set the siteUrl in  `config/general.php` as it depends on the enivronment it's being hosted in (local, dev, stage, production):

```php
<?php
/**
 * General Configuration
 *
 * All of your system's general configuration settings go in here. You can see a
 * list of the available settings in vendor/craftcms/cms/src/config/GeneralConfig.php.
 */
return [
    // Base site URL
    'siteUrl' => (array_key_exists('HTTPS', $_SERVER)
        ? 'https'
        : 'http') . '://' . $_SERVER['HTTP_HOST']
]
```

Previously in Craft v2 we would set this as a constant via `CRAFT_SITE_URL`, but that constant has been [deprecated](https://github.com/craftcms/cms/blob/1967f4d87ea506fae557254554890fb2ea33d43d/src/services/Config.php#L122) and the reason we moved it to `config/general.php`.

I went ahead and made a pull request with what I think is the best solution at https://github.com/Dolphiq/craft3-plugin-sitemap/pull/8, hopefully it's something that can be merged in. Either way, I'd appreciate any feedback on it.
